homeassistant/components/shelly/sensor.py

```diff
@@ -355,21 +391,27 @@ async def async_setup_entry(
 class BlockSensor(ShellyBlockAttributeEntity, SensorEntity):
     """Represent a block sensor."""
 
+    entity_description: BlockSensorDescription
+
+    def __init__(
+        self,
+        wrapper: BlockDeviceWrapper,
+        block: Block,
+        attribute: str,
+        description: BlockSensorDescription,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(wrapper, block, attribute, description)
+
+        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
+        if unit_fn := description.unit_fn:
+            self._attr_native_unit_of_measurement = unit_fn(block.info(attribute))
```

Comment:
Moved out of `ShellySleepingBlockAttributeEntity` as `unit_of_measurement` doesn't apply outside sensor platform